rpcs3/Emu/Cell/Modules/sys_net.cpp

```diff
@@ -148,57 +177,116 @@ namespace sys_net
 		return g_tls_net_data.ref(&_tls_data_t::_h_errno);
 	}
 
+	// Error helper functions
+	s32 get_last_error()
+	{
+		// Convert the error code for socket functions to a one for sys_net
+		s32 result;
+		const char* name{};
+
+#ifdef _WIN32
+		switch (s32 code = WSAGetLastError())
+#define ERROR_CASE(error) case WSA ## error: result = SYS_NET_ ## error; name = #error; break;
+#else
+		switch (s32 code = errno)
+#define ERROR_CASE(error) case error: result = SYS_NET_ ## error; name = #error; break;
+#endif
+		{
+			ERROR_CASE(EWOULDBLOCK);
+		default: throw fmt::exception("Unknown/illegal socket error: %d" HERE, code);
+		}
+
+		if (name && result != SYS_NET_EWOULDBLOCK)
+		{
+			ppu_error_code::report(result, name);
+		}
+
+		return result;
+#undef ERROR_CASE
+	}
+
 	// Functions
 	s32 accept(s32 s, vm::ptr<sockaddr> addr, vm::ptr<u32> paddrlen)
 	{
 		libnet.warning("accept(s=%d, family=*0x%x, paddrlen=*0x%x)", s, addr, paddrlen);
-		s = g_socketMap[s];
+		socket_t sock = idm::get<sys_net_socket>(s)->s;
 
-		if (!addr) {
-			int ret = ::accept(s, nullptr, nullptr);
-			get_errno() = getLastError();
-			return ret;
+		s32 ret;
+
+		if (!addr)
+		{
+			ret = ::accept(sock, nullptr, nullptr);
+
+			if (ret < 0)
+			{
+				libnet.error("accept(): error %d", get_errno() = get_last_error());
+				return -1;
+			}
 		}
-		else {
+		else
+		{
 			::sockaddr _addr;
-			memcpy(&_addr, addr.get_ptr(), sizeof(::sockaddr));
-			_addr.sa_family = addr->sa_family;
-			::socklen_t _paddrlen;
-			s32 ret = ::accept(s, &_addr, &_paddrlen);
+			::socklen_t _paddrlen = 16;
+
+			ret = ::accept(sock, &_addr, &_paddrlen);
+
+			if (ret < 0)
+			{
+				libnet.error("accept(): error %d", get_errno() = get_last_error());
+				return -1;
+			}
+
 			*paddrlen = _paddrlen;
-			get_errno() = getLastError();
-			return ret;
+			addr->sa_len = _paddrlen;
+			addr->sa_family = _addr.sa_family;
+			memcpy(addr->sa_data, _addr.sa_data, addr->sa_len - 2);
 		}
+
+		return idm::make<sys_net_socket>(ret);
 	}
 
 	s32 bind(s32 s, vm::cptr<sockaddr> addr, u32 addrlen)
 	{
 		libnet.warning("bind(s=%d, family=*0x%x, addrlen=%d)", s, addr, addrlen);
-		s = g_socketMap[s];
+		socket_t sock = idm::get<sys_net_socket>(s)->s;
```

Comment:
You didn't really fix anything. The result of get() call must be saved in local variable.